pkg/operator/resources/operator/operator.go

```diff
@@ -528,6 +527,40 @@ func createCDIListCRD() *extv1beta1.CustomResourceDefinition {
 				ShortNames: []string{"cdi", "cdis"},
 			},
 
+			Validation: &extv1beta1.CustomResourceValidation{
+				OpenAPIV3Schema: &extv1beta1.JSONSchemaProps{
+					Properties: map[string]extv1beta1.JSONSchemaProps{
+						"apiVersion": {
+							Type: "string",
+						},
+						"kind": {
+							Type: "string",
+						},
+						"metadata": {
+							Type: "object",
+						},
+
+						"spec": {
+							Properties: map[string]extv1beta1.JSONSchemaProps{
+								"imageRegistry": {
+									Type: "string",
+								},
+								"imageTag": {
+									Type: "string",
+								},
+								"imagePullPolicy": {
+									Type: "string",
+								},
+							},
+							Type: "object",
+						},
+					},
+					Required: []string{
+						"spec",
```

Comment:
Not sure spec should be required if all spec params are optional

Reply:
> Not sure spec should be required if all spec params are optional

ImagePullPolicy param mentioned as required in types.go  https://github.com/kubevirt/containerized-data-importer/blob/b3bf364b238217750be5bd0d1c499630fbfb751b/pkg/apis/core/v1alpha1/types.go#L199

So we can mark ImagePullPolicy as required in CRD validation section and also I think Spec is mandatory root entry for CRD.

Reply:
No, ImagePullPolicy should required.  `valid:"required"` means that the value (if set) cannot be empty string

spec should not be required until it has a member that is required.

Reply:
okay, In that case, I will remove the required from Spec.

Reply:
removed. the required from Spec.